### ISSUE TITLE
[2][diegogomes] UX for Price Alerts

### ISSUE THREAD

**Issue by wmbutler** [bitshares/bitshares-ui#1263]
Please UX for setting the following alerts:

- [ ] Market price value increase / decrease
- [ ] Market percentage increase / decrease
- [ ] Collateral ratio target

Related to: #422

**Comment by happyconcepts** [comment]
*Alerts are by nature focused on the future*. Please consider whether displaying the alert notification history (last 5 was mentioned in #422 ) is in fact desirable. My reservation is that it potentially clutters the UI (and hence the UX) without adding benefit.

Very happy to see progress on this vital feature.

**Comment by diogogomes** [comment]
@happyconcepts,

I am planning to present more than one possibility even today.
Can you explain more what you mean with:
> My reservation is that it potentially clutters the UI (and hence the UX) without adding benefit.

Thanks

**Comment by diogogomes** [comment]
@wmbutler, @gibbsfromncis, @happyconcepts, please give me your feedback.

**Notes:**

I created this UI/UX only for the items mentioned but always keeping in mind new additions in the future (ex: transfers notifications, filled orders, etc.)

Market price/percentage value increase / decrease is based on user starred assets (this way,we can implement with the code that we already have)

All colors, typefaces, etc are susceptible to modifications.

I used the light theme from staging branch as a base.

Although I mentioned in the issue #442 using tabs inside the notifications dropdown, I decided not to do to make it easier to view by the user.

I have created several icon alternatives from notifications and drop-down locations.

---

## Notification icon
   
### Alternative 1
![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/notification%20alt1%20empty.png) ![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/notification%20alt1%20filled.png)

### Alternative 2
![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/notification%20alt2%20empty.png) ![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/notification%20alt2%20filled.png)

### Alternative 3
![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/notification%20alt2%20empty.png) ![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/notification%20alt3%20filled.png)

### Alternative 4
![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/notification%20alt4%20empty.png) ![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/notification%20alt4%20filled.png)

## Notification Dropdown

I've included a direct access icon to the notifications settings and to all recent activity (keeping in mind the addition of the activity, in the future).

### Alternative 1

Aligned with the notifications icon and with a small triangle for visual help

![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/dropdown%20alt1.png)

### Alternative 2

Aligned with the burger menu and with a small triangle for visual help

![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/dropdown%20alt2.png)

### Alternative 3

Aligned with the burger menu taking the approach of the actual version

![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/dropdown%20alt3.png)

### Alternative 4

Aligned with the notifications icon taking the approach of the actual version

![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/dropdown%20alt4.png)

---

@wmbutler, please tell me which one you prefer and if I should upload it or make a pull request with the measurements I used. Thanks

**Comment by wmbutler** [comment]
- I like Alternative 3 for the notification icon.
- I tend to prefer a menu for the dropdown so none of those options are exactly what I'd expect regarding the dropdown.

**Comment by diogogomes** [comment]
@wmbutler I'm sorry to hear that you don't like the drop-downs. I assumed it was to take the direction I had described in the issue #422.

Can you please give a suggestion in another direction, so that I may propose other alternatives?

Thanks

**Comment by wmbutler** [comment]
I just meant that the menus should be connected to the header instead of having a gap or a carat.

**Comment by diogogomes** [comment]
Sorry @wmbutler , I forgot to mention in the previous description that I integrated the concept of Dropdown with Popup.

I'll make an alternative with the notifications as a simple drop-down menu.

**Comment by diogogomes** [comment]
@wmbutler can you review, please?

## Notification Menu Dropdown

I thought in another mode for the notifications icon when the Drop-down is open:

### Alternative 5

With header and right aligned with the notifications icon

![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/dropdown%20alt5.png)

### Alternative 6

With header and right aligned with the burger menu

![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/dropdown%20alt6.png)

### Alternative 7

With header and right aligned with the border of the *content*

![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/dropdown%20alt7.png)

### Alternative 8

Without header and right aligned with the notifications icon

![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/dropdown%20alt8.png)

### Alternative 9

Without header and right aligned with the burger menu

![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/dropdown%20alt9.png)

### Alternative 10

Without header and right aligned with the border of the *content*

![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/dropdown%20alt10.png)

**Comment by wmbutler** [comment]
I see arguments for both 8 and 10. I do think that the selected background state for the bell should match the background color of the menu.

**Comment by happyconcepts** [comment]
> Can you explain more what you mean with:
> My reservation is that it potentially clutters the UI (and hence the UX) without adding benefit.

I would like to have the option that alerts disappear once I have read them. Moreover, I think it should be the default behavior; user would have to opt-in to keep and display alert history.

Because already I have to deal with two time states, future and now: alerts that are set but have not triggered yet are in the future, and alerts that have triggered are now. If I have to wade through the litany of my past alert history all the time just to be sure of what's happening now...then I would feel discouraged that there is an alert feature and I cannot use it because it's too confusing.

I hope that describes it a little; more of a focus on the future and of course, the moment when an alert condition happens, instead of focus on the past. We can always go to history or the blockchain itself to view the past.

<hr>
As for the designs, my only complaint is the red color of the circle. To me it should not be red or green, at least here in the U.S. because those colors are associated with up and down price movement (I know it's different in China, generally). 

Will you try using White, or Yellow, or a shade of Blue for the circle?

**Comment by diogogomes** [comment]
@wmbutler 

> I see arguments for both 8 and 10.

Personally, I prefer the dropdown right aligned with the icon.

This allows a better visual connection between the icon and its dropdown

#

> I do think that the selected background state for the bell should match the background colour of the menu.

You are totally right.

I tried to match with all the bell but it becomes too similar to the background of the navigation bar (can be easily confused with the disabled state).

What's your opinion about matching the interior colour of the bell keeping the borders colour?

![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/dropdown%20alt8.png)

**Comment by diogogomes** [comment]
> As for the designs, my only complaint is the red color of the circle. To me it should not be red or green, at least here in the U.S. because those colors are associated with up and down price movement (I know it's different in China, generally).
Will you try using White, or Yellow, or a shade of Blue for the circle?

@happyconcepts,

Thank you for the explanation and input.

In Europe, the red can have same meaning of US but it's also an **warning colour** (like traffic signs).

The reason why I choose red it's because red is the **complementary colour** of blue and also have a warning meaning.
 
Even so, I decided to elaborate more alternatives:

![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/icon%20alt8.2.png)![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/icon%20alt8.3.png)
![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/icon%20alt8.4.png)![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/icon%20alt8.5.png)

**Comment by gibbsfromncis** [comment]
I suppose red circle on Bell is a good color to notify user about new notifications. I don't see any reasons why it should be any another color. 

Another option is to remove red circle at all and just play with Bell background like make it filled when notifications are and make it 30% of opacity when no new notifications.

**Comment by gibbsfromncis** [comment]
@happyconcepts I'm so sorry if I said something wrong but please keep calm. I don't think the red color is a signal of price movement totally. I suppose it's personal. I didn't propose to use red arrow.  

I'm partially color blind too so I always think about blink people 😄 

@diogogomes if Bill disagree with Red circle too please before change the color of Circle add 2-3px border to it like a panel background to visually split Bell and Circle. This should help users to visually identify they have new notifications

**Comment by diogogomes** [comment]
@ahdigital, can you give your opinion regarding the light theme and your style guide? Thanks

**Comment by wmbutler** [comment]
@diogogomes please work with @ahdigital on UX acceptance. I think he is out for the next few days.

**Comment by diogogomes** [comment]
@ahdigital, can you give me your opinion taking into account the light theme and the UI Style Guide? Thank you

**Comment by ahdigital** [comment]
In my mock ups I've been using this:

![image](https://user-images.githubusercontent.com/3088454/37584099-06bb6f40-2b4c-11e8-838a-16ac72f6e3ee.png)

I don't have an issue with using a red circle as the identifier (without count) for new notifications as it's a common design pattern to do so. Combined with a bell icon the expectation of functionality should be clear.

The consideration raised about timeliness is important. I think a pattern whereby once the notification is considered 'read' it is removed from the drop down*, is a good one. The question is, how do we consider it to be 'read'? I'd like to have the notification itself clickable and it takes you somewhere useful rather than having buttons to mark them 'read'/'unread'.

\* The user should be able to review 'read' notifications somehow, even if it's a limited history; E.g. 30 days.

There are 3 use cases for notifications mentioned in this issue, but there will be scope to include more. Might we run into issues with say 6 types and we need an icon for each?

**Comment by wmbutler** [comment]
A suggestion:

- Clicking on the bell notification marks everything as read.
- All items remain in the list paged 25 at a time

**Comment by ahdigital** [comment]
> Clicking on the bell notification marks everything as read

And then they are removed from the drop down?

**Comment by gibbsfromncis** [comment]
@ahdigital the idea is to use like on a facebook - you just click and red circle disappears and that's it. You can anytime open dropdown by click on the bell and see the previous notifications.

**Comment by gibbsfromncis** [comment]
@wmbutler would you like to use notifications for price alerts only? I suppose it would be good to use it for any activity. Also it would be helpful to notify user about expired trades (https://github.com/bitshares/bitshares-ui/issues/1331#issuecomment-374294006)

**Comment by wmbutler** [comment]
I think we build it into the browser notifications.

**Comment by gibbsfromncis** [comment]
@wmbutler sure. But we will keep bell icon on header, right? If yes, what kind of notifications are we going to display there?

**Comment by wmbutler** [comment]
I would suggest that we keep a master list of all notifications we intend to display and ensure that when we add a new notification, we make sure it gets handled at the bell icon and, if the user has browser notifications, at the browser level. The user will have the ability to activate and deactivate all notifications that we have chosen to implement.

This is what is in the current client:

![screen shot 2018-03-24 at 8 20 21 pm](https://user-images.githubusercontent.com/1254810/37870645-d1d46706-2fa0-11e8-9b3e-4128e8b456ea.png)

**Comment by gibbsfromncis** [comment]
@wmbutler Got it. When you approve UX let me know about start of code part please. Also I would like to update estimate of code part task. 

@diogogomes Please do not forget to provide some controls to add/remove notifications for pair of assets. For now I see only how to display Price Alerts but I do not understand where user can create/remove alerts.

**Comment by wmbutler** [comment]
@diogogomes also, make sure you follow the (in progress) style guide being fostered by @gibbsfromncis and @ahdigital

**Comment by diogogomes** [comment]
@gibbsfromncis 

> For now I see only how to display Price Alerts but I do not understand where user can create/remove alerts.

The notifications are based on starred pairs. Can you give your opinion about the code implementation referred on previous comment?

- Market price/percentage value increase / decrease is based on user starred assets (this way, we can implement with the code that we already have)

---

@gibbsfromncis and @ahdigital can you provide the link for the in progress style guide? (I searched on the new style guide repo but it's empty.) Thanks

**Comment by wmbutler** [comment]
We would need UX for setting the percentage or fixed increase/decrease right?

**Comment by diogogomes** [comment]
> We would need UX for setting the percentage or fixed increase/decrease right?

I was thinking in a daily alert (like the portfolio tracking apps). @wmbutler what you think is the best approach?

**Comment by wmbutler** [comment]
I think a user would want to set an alert for a 20% price increase or decrease. Or a fixed value. Why would I want a daily alert?

**Comment by diogogomes** [comment]
okay @wmbutler, what is the best approach? A new icon on side of the pair and a pop up modal with configurations when clicked or configuration options together with the general settings notifications?

**Comment by wmbutler** [comment]
I'm not the UX guy here :) you might consult @ahdigital. Happy to add some additional time for this if needed. Fundamentally, I think it makes a lot of sense to set price alerts in starred markets since this would be a definitive list of markets that are of interest to a user.

Probably a control here that triggers a modal.

<img width="1072" alt="screen shot 2018-03-26 at 5 06 35 pm" src="https://user-images.githubusercontent.com/1254810/37935689-193a9534-3118-11e8-859c-3d0973082993.png">

**Comment by diogogomes** [comment]
Thanks, my opinion is the same. I think it's better on starred markets. I'll provide UI for this approach (modal).

**Comment by ahdigital** [comment]
I think it'd be logical to allow users to setup price alerts from the market page. It's a tool closely related to the subject of this page and brings the functionality to the surface. It could be triggered from:

**Summary bar**
![image](https://user-images.githubusercontent.com/3088454/37957198-412aa92a-31a5-11e8-9601-c6fd4465a3bf.png)

 **Or, my markets**
![image](https://user-images.githubusercontent.com/3088454/37957252-62f7d14a-31a5-11e8-9766-f21cedbcff17.png)


Having it on starred markets is great and I think it should be there as well, but I see the use case as more of a broad overview where adjustments can be made and not the first step in the flow.

@diogogomes The in progress style guide is [here](https://www.behance.net/gallery/62549753/Bitshares-DEX-UI-style-guide).

**Comment by diogogomes** [comment]
##### (@wmbutler, sorry for the delay but without an editable style guide or docs it was more time consuming to try to match the correct measurements, margins, etc)

Thank you for all opinions and ideas.

In the process of adapting the @ahdigital Style Guide, I tried to implement the most of your ideas.

---

## Icon without new notifications

![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/(final%20candidate)%20without%20notifications.png)

---

## Icon with new notifications

![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/(final%20candidate)%20with%20notifications.png)

---

## Notifications opened

![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/(final%20candidate)%20dropdown.png)

---

## Starred/featured markets alert clickable icon (to open the modal)

![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/featured_starred%20markets%20alert%20icon.png)

---

## Exchange alert clickable icon (to open the modal)

![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/exchange%20alert%20icon.png)

---

## Modal

![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/modal.png)

**Comment by wmbutler** [comment]
I think your UX will need to incorporate a percentage change up/down or fixed value.

**Comment by diogogomes** [comment]
@wmbutler, are you referring to the possibility of setting the alert in the modal with respect to the variation in percentage?

**Comment by gibbsfromncis** [comment]
@diogogomes please add some toggle or radio buttons where user be able to change type of alert from fixed to % and vice versa. 

Also I suppose we need two fields for fixed values like:
- Notify if value more than: 10.000 bitUSD / BTC
- Notify if value less than 5.000 bitUSD / BTC

**Comment by wmbutler** [comment]
I'd consolidate this into a single input field with a separate control to indicate what type of alert. you are looking for.

**Comment by clockworkgr** [comment]
Don't think it needs a separate control to select.

Assuming you have some sort of gauge/number picker with up/down controls, if you start at current price, whether you want above or below is decided by the direction in which you place the alert.

Assume BTC price is at 10000USD...if you move the alert threshold to 11000, it's obviously an above alert (below would trigger immediately).

Same goes if you moved the threshold to 9000 (where above would trigger immediately and thus is obviously a below trigger).

Example from coinbase's alerts (where above or below glows to indicate type of alert automatically depending on where you move the gauge to compared to current price):
![image](https://user-images.githubusercontent.com/6826762/38238818-bbd9a26e-3733-11e8-9d38-83ef217be974.png)

**Comment by diogogomes** [comment]
Hi @clockworkgr, Nice to see your opinion. I appreciate it, thanks.

# 

@wmbutler,

Given the previous opinions and after I have detected that there are two distinct types of needs in relation to alerts, I decided to create a version for multi alerts.

Thus, we can meet the needs of users like @gibbsfromncis who want more than one price alert and maintaining the simplicity to not confuse the less advanced users.

In the drop-down, I fulfilled the UI Style Guide of @ahdigital.

Please give me your opinion. Thank you.

---

## Modal with the first field (drop-down open)

![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/modal1.png)

---

## Modal with more than one alert

- After click on `Add new` appears a new field.

- If clicked on red button, the field disappears until stay only one field that can be cleaned clicking on the respective red button.

- There is only one option for the price (fixed price). The alert activates when it reaches this value, whether it is going up or down the price.

- Can be set multiple prices at same time.

![](https://gitlab.com/diogogomes.eu/bitshares-ui-contributions-shared/raw/iss1263/modal2.png)

**Comment by gibbsfromncis** [comment]
@wmbutler please comment this. This one blocks my #422

**Comment by happyconcepts** [comment]
Heading of Alert confuses me if this is the screen where I set the alerts. From that heading I expect to see a list of my alerts?

Alerts must be stored locally. Why is there a confirm button...or am I confused that setting an alert requires me to pay gas?

**Comment by diogogomes** [comment]
@happyconcepts, the alerts are set in starred/featured markets alert icon and in exchange alert icon. The alerts can be viewed in the modal.

About been stored locally or cross platform and if is required pay fees, It's a good question and is better discuss this in the issue #422 (the implementation). Can you give your opinion there? Thank you

**Comment by diogogomes** [comment]
Hello @wmbutler, I finished this issue at 12 April. Any reason why you changed the milestone? Thank you.

**Comment by wmbutler** [comment]
Because the UX is just not good. I'm sorry to be so blunt but isn't it obvious?

**Comment by diogogomes** [comment]
Sorry and with all respect, it isn't obvious without any kind of feedback.

I'm just complying with the @ahdigital User Interface Style Guide.

I have other suggestions but it requires a big changes to the UI Style Guide and because you request to comply with this type of UI, I'm frozen with this type of suggestions and I can't take a more friendly User Interface path.

Bringing the private conversation to here (because your response on GitHub was too ambiguous without a path to follow):

@wmbutler 
> I’m having a hard time with comments because the UX is still confusing to me. As a user I don’t understand why I’d have 3 different options on the same page rather than a single option with 3 different criteria.

@diogogomes
> I was trying to bring @gibbsfromncis suggestion to have more than one alert at same time.
>
>~~Really late here, tomorrow I will check your answer on github.~~
>
>I need to have a strong opinion from you @billbutler and @ahdigital, you are the team leader and Dominic UX expert.
>
>It's difficult, confusing an frustrating lose time try to accomplish all opinions and do not comply with your vision.
>
>I prefer comply with your vision and deliver the work done

What path/direction you want to take?

Thank you

**Comment by wmbutler** [comment]
If I have to design the UX, why would we offer a bounty for it?

**Comment by diogogomes** [comment]
> If I have to design the UX, why would we offer a bounty for it?

I'm sorry but I honestly do not think that this type of answers (like the one where you tell me to fork the project) are professional, productive and I do not believe they contribute to a good course in the development of Bitshares UI.
To this type of answer, I just have to say that no one suggested that you do the Design/User Interface, just ask for a more communicative leadership and a decision what you want me to do.

I suggested only one alert in modal (of each type - 3 options), it was suggested to be more than one in simultaneous.

I suggested more than one alert where only the first field appears and when you click "Add new " the next field appears, it was suggested to be only one field with 3 options.

I don't like to go in circles because I don't think it's professionally feasible.

In short, you don't like one field and you don't like more than a field.

Do you have any opinions, or may I suggest mine without constantly being with detours around my work? If it is necessary, I will take the decision and assume the responsibility.

**Comment by gibbsfromncis** [comment]
@wmbutler I'm sorry but I want to take a part of your discussion and put my thoughts there.

@diogogomes I suppose there are no reasons to continue this discussion. You just didn't deliver the UI/UX which make us say `Wooow`. Honestly, I agree with Bill for 100%. When we say we don't like it - you ask us for a direction you should go. We're there not to teach you or spend our time for non-profit discussions. If you are not ready to delivery things we like - it's OK. Maybe our expectations so high. Please don't teach us to do our job or say that somebody do something unprofessional. We are open community and we should respect each other. 

I would like to propose you just to be professional and take that your solution doesn't fit our requirements, time of this issue expired and just ask us to unassign you from this issue.

**Comment by diogogomes** [comment]
@wmbutler, you removed my 4 hours when @ahdigital break the follow rule on issue #1030.

> If an issue is already claimed (assigned), do not attempt to claim it. Issues claimed by outside developers will indicate an assignment to wmbutler, but will mention the developer's github account in this the comments.

Now, you let the date expire without feedback to not pay me again breaking another rule.

I'll be paid, one way or another and I'm glad to see this situations on public domain.

Thank you for your time.
